File: backend_original/src/main/java/com/example/dealshare/project/model/Product.java
package com.example.dealshare.project.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@Table(name ="product")
@NoArgsConstructor
@AllArgsConstructor
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name ="id")
    private int id;
    @Column(name ="name")
    private String name;
    @Column(name ="description")
    private String description;

    @Column(name ="category_id")
    private int category_id;

    @Column(name = "price")
    private int price;
    @Column(name ="photo")
    private String photo;

}

File: frontend/app/app/src/main/java/com/example/app/Product.kt
package com.example.app

import com.google.gson.annotations.SerializedName

data class Product(
    @SerializedName("id")
    var id: Int,
    @SerializedName("name")
    var name: String? = null,
    @SerializedName("description")
    var description: String? = null,
    @SerializedName("category_id")
    var category_id:Int?=null,

    @SerializedName("price")
    var price: String? = null,

    @SerializedName("photo")
    var photo: String?=null,
)